agent/internal/container.go

```diff
@@ -108,7 +108,12 @@ func (c *containerActor) Receive(ctx *actor.Context) error {
 			ctx.Tell(c.docker, pull)
 			c.baseTaskLog = taskLog
 		} else {
-			ctx.Tell(c.docker, reattachContainer{ID: c.Container.ID})
```

Comment:
Am I understanding this race right? 

the container actor sends ```reattachContainer``` to the docker actor 
which can send ```containerReattached``` back to the container actor setting ```containerInfo```

Then there was no synchronization between using ```containerInfo``` and that ```containerReattached``` message arriving?

Reply:
`aproto.SignalContainer` needs the `containerInfo`, so if it arrives before `containerReattached` it'd hit nil pointer. so instead we try to get that data more "synchronously", which is a mess in the actor system.
for example, it may happen when agent has disconnected and master had to buffer some messages and then resend them when agent is back.

Reply:
writing this comment made me realized we should also ping the docker actor for good measure.

Reply:
re-reading, i'm actually don't get how this helps us get containerInfo any better? if the container actor receives a signal before `containerReattached` is sent back after the docker actor prestart, don't we still panic in the container actor trying to access c.containerInfo.id?

Reply:
it seems this would make it less likely but not impossible, since it basically saves a round trip through the actors

Reply:
it we had actually synchronization (reattachContainer as an actor ask returning containerInfo) between starting the container and having it reattached with container info, then it would work?

Reply:
I think we need one more ping: `containers` actor should ask reattached `container` actors.